assembly/src/tokens/mod.rs

```diff
@@ -308,15 +309,5 @@ fn validate_proc_invocation_label<'a>(
 /// A label is considered valid if it start with a letter and consists only of letters, numbers,
 /// and underscores.
 fn is_valid_label(label: &str) -> bool {
-    // a label must start with a letter
```

Comment:
I think we should remove the `is_valid_label(..)` function entirely and replace it's invocations with direct calls to `PROCEDURE_LABEL_PARSER.parse_label(label.to_string())`.  It should be noted that currently `is_valid_label(..)` is used to parse both procedure labels and module labels.  For the minute it is likely valid to use `PROCEDURE_LABEL_PARSER` for both procedure labels and module labels.  However, if we do this then we should rename the parser to something like `GENERIC_LABEL_PARSER` and then we will have other specialised parsers with different rule sets (e.g. `CONSTANT_LABEL_PARSER`).

Reply:
@Danijel-Enoch I will second @frisitano's recommendation. He's more familiar with this part of the code and has made a good point about the usage of `is_valid_label` for module labels. Can you take a look at this approach?